test/testutil/fake_random.c

```diff
@@ -179,8 +196,9 @@ OSSL_PROVIDER *fake_rand_start(OSSL_LIB_CTX *libctx)
         return NULL;
 
     /* Ensure that the fake rand is initialized. */
-    if (!TEST_ptr(RAND_get0_private(libctx))
-            || !TEST_ptr(RAND_get0_public(libctx))) {
+    if (!TEST_true(check_rng(RAND_get0_primary(libctx), "primary"))
```

Comment:
why does it care about primary?

Reply:
If somebody ever accesses the primary directly, the link and the name need to be setup or the callback won't work.  It's a harmless precaution against something changing in the future.